### ISSUE TITLE
[BUG] daemon and tray silently fail to load as part of setup if `launchctl remove` used

### ISSUE THREAD

**Issue by praveenkumar** [code-ready/crc#1140]
### General information

  * OS:  macOS 
  * Hypervisor: hyperkit
  * Did you run `crc setup` before starting it (Yes/No)? yes

If a user use `launchctl remove crc.daemon` or `launchctl remove crc.tray` then setup command is not able to start those launch agents without any error or warning.

**Comment by gbraad** [comment]
> start those launch agents without any error or warning.

not sure how this should be read. Can you show the error in some way... or you suggest there is no error, and therefore silently fails?

**Comment by praveenkumar** [comment]
> not sure how this should be read. Can you show the error in some way... or you suggest there is no error, and therefore silently fails?

If you try following, you expect to have tray icon and daemon running after the setup command which is not happening and also there is no error from the setup command which checks for the daemon and tray.

```
$  launchctl remove crc.daemon
$  launchctl remove crc.tray

$ ./crc setup --enable-experimental-features --log-level debug
INFO Checking if oc binary is cached              
DEBU oc binary already cached                     
INFO Checking if podman remote binary is cached   
DEBU podman remote binary already cached          
INFO Checking if goodhost binary is cached        
DEBU goodhost remote binary already cached        
INFO Checking if CRC bundle is cached in '$HOME/.crc' 
INFO Checking if running as non-root              
INFO Checking if HyperKit is installed            
DEBU Checking if hyperkit is installed            
INFO Checking if crc-driver-hyperkit is installed 
DEBU Checking if crc-driver-hyperkit is installed 
DEBU Running '/Users/prkumar/.crc/bin/crc-driver-hyperkit version' 
DEBU crc-driver-hyperkit is already installed in /Users/prkumar/.crc/bin/crc-driver-hyperkit 
INFO Checking file permissions for /etc/resolver/testing 
INFO Checking if tray binary is installed         
INFO Checking if launchd configuration for daemon exists 
INFO Checking if launchd configuration for tray exists 
INFO Checking installed tray version              
INFO Checking if CodeReady Containers daemon is running 
INFO Check if CodeReady Containers tray is running 
Setup is complete, you can now run 'crc start -b $bundlename' to start the OpenShift cluster
```

**Comment by gbraad** [comment]
> > silently fails

> there is no error 

👍

**Comment by anjannath** [comment]
So this is due to two things..
1. When `launchctl remove ...` is ran `launchctl list crc.tray` doesn't provide any o/p and the `launchd.AgentRunning()` fails to determine correctly if an agent is running, so the fix functions are never triggered.
2. Even when fix functions are ran, `launchctl start` wouldn't work as it needs the plist files already loaded to start, which was removed/unloaded by the `launchctl remove` command.

**Comment by praveenkumar** [comment]
Resolved by #1142
